### PR TITLE
feature/ARC 522 add sbt build step

### PR DESCRIPTION
**Pull request by koeves**

### CODE REVIEW

File: src/mpyl/steps/build/sbt.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            logger=logger,
+            meta=Meta(
+                name='Sbt Build',
+                description='Build sbt project',
+                version='0.0.1',
+                stage=Stage.BUILD
+            ),
+            produced_artifact=ArtifactType.DOCKER_IMAGE,
+            required_artifact=ArtifactType.NONE,
+            after=None  # AfterBuildDocker(logger=logger)
```
Comment (koeves) [inline src/mpyl/steps/build/sbt.py:25]: Use `AfterBuildDocker`

Comment (koeves) [reply src/mpyl/steps/build/sbt.py:25]: Explain this in readme

File: src/mpyl/utilities/sbt/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"""SBT config"""
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class SbtConfig:
```
Comment (koeves) [inline src/mpyl/utilities/sbt/__init__.py:6]: Move this to config

File: src/mpyl/steps/build/sbt.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        )
+
+    def execute(self, step_input: Input) -> Output:
+        with open(SbtConfig.java_opts_file_name, 'w+') as jvm_opts:
+            jvm_opts.write(SbtConfig.sbt_opts.replace(' ', '\n'))
+            check_fmt: Optional[str] = \
+                'scalafmtCheckAll' if step_input.run_properties.target == Target.PULL_REQUEST else None
+            image_name: str = docker_image_tag(step_input)
+            commands: list[str] = [
```
Comment (koeves) [inline src/mpyl/steps/build/sbt.py:34]: Move this to a helper

Comment (koeves) [reply src/mpyl/steps/build/sbt.py:34]: Reuse `custom_check_output` from help deploy, move to utilities

File: src/mpyl/steps/deploy/k8s/helm.py [hunk 4]
```diff
@@ -68,7 +51,7 @@ def install(logger: Logger, step_input: Input, name_space: str, kube_context: st
 
     if step_input.dry_run:
         cmd = f"helm upgrade -i {chart_name} -n namespace --kube-context {kube_context} {chart_path} --debug --dry-run"
-        return custom_check_output(logger, cmd)
+        return custom_check_output(logger, cmd.split(" "))
 
     cmd = f"helm upgrade -i {chart_name} -n {name_space} --kube-context {kube_context} {chart_path}"
-    return custom_check_output(logger, cmd)
+    return custom_check_output(logger, cmd.split(" "))
```
Comment (SamTheisens) [inline src/mpyl/steps/deploy/k8s/helm.py:57]: Why do you want to do the command splitting on the outside?

Comment (koeves) [reply src/mpyl/steps/deploy/k8s/helm.py:57]: In sbt build's case I have to supply the DOCKER_BUILDKIT env var as well, which made it a bit cumbersome to do with splitting by default.. But I can take a look to improve this

Comment (SamTheisens) [reply src/mpyl/steps/deploy/k8s/helm.py:57]: Ah, right. I see. 🤔 maybe we could support two alternative apis? Or use a union type that accepts a string or a list of strings.

**Review by ewaldschrader — APPROVED** [review]

**Review by koeves — COMMENTED** [review]
Nice improvements!!

File: tests/steps/build_step/sbt/test_sbt.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from src.mpyl.steps.build.sbt import BuildSbt
+from src.mpyl.steps.models import Input
+from tests.test_resources import test_data
+from tests.test_resources.test_data import get_project
+
+
+class TestBuildSbt:
```
Comment (koeves) [inline tests/steps/build_step/sbt/test_sbt.py:7]: In the future we could convert these tests to use the [unittest](`unittest`) library, like what DA is using in their scripting repo, if we feel like we need more facilities for testing

File: src/mpyl/utilities/sbt/__init__.py (hunk 2 above)
Comment (koeves) [reply src/mpyl/utilities/sbt/__init__.py:6]: This should still be attempted I think

File: tests/steps/build_step/sbt/test_sbt.py (hunk 5 above)
Comment (SamTheisens) [reply tests/steps/build_step/sbt/test_sbt.py:7]: Oh, I actually moved away from the standard  python `unittest` library :-) https://github.com/Vandebron/mpyl/pull/12

Comment (koeves) [reply tests/steps/build_step/sbt/test_sbt.py:7]: Ohh I should've checked.. Then disregard my comment!

**Pull request merged**